keras_cv/layers/preprocessing/rgb_shift.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np 
+import tensorflow as tf
+from tensorflow.keras import layers, backend
+
+
+class RGBShift(layers.Layer):
+    """RGBShift class randomly shift values for each channel of the input RGB image. The expected images 
+    should be [0-255] pixel ranges.
+
+   
+    Input shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+    Output shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+
+    Args:
+        r_shift_limit: A scalar or tuple or list of two upper and lower bound number. It will change r-channel by
```

Comment:
- Is it really worth splitting the shift argument into 3? Most use cases will want to use the same value for all channels, and having to respecify it 3 times add significant UX friction.
- This type of argument is normally called something like `factor` is other preprocessing layers. We should align on existing naming covnentions. https://keras.io/api/layers/preprocessing_layers/image_augmentation/
- Likewise we should align on the same semantics as with other preprocessing layers: an integer is used as a level number, a float is used as a fraction of the total (e.g. `0.1` is interpreted as "shift within a +-10% range")

I'd recommend a single argument `factor` with the semantics above. To cover the use case of per-channel factors (which is rare), we could simply add a `target_channel=None` argument so you can specify which channel(s) the shift should apply to (default to all). What do you think?

Reply:
@fchollet 
Regarding the shift value to the RGB channel, I don't have empirical evidence to use shift values independently to all channels, but I think it's nice to have an option.  Also, independent scaling to all channels looks quite different or randomized than the original channel whereas single scaling to all channels more and more looks like a brightness adjusted. That's why I think it's fine to have an option for independent channel shifting.  So, using `target_channel`, we can prove an option to the end-user to pick which one suits their goals. Like, 

```python
(pseudo)

if target_channel == 0:  # R channel 
    # only R channel will be shifted and concat with others 

if target_channel == 1:  # G channel 
    # only G and concat with others 

if target_channel == 2:  # B channel 
    # only B and concat with others 

if target_channel == "RGB":
    # shit independently to all channel and concat all 
```

WDYT?

Reply:
@fchollet 

Regarding the `factor` param, are you suggesting the following setup? 

```python

import tensorflow as tf

factor = (-0.6, 0.6)
shift_limit = tf.random.uniform([], 
                                minval=factor[0],
                                maxval=factor[1], 
                                dtype=tf.float32)*255.0
shift_limit
<tf.Tensor: shape=(), dtype=float32, numpy=-7.2357845>
```

Reply:
Upon reflection I'd recommend dropping the argument altogether -- just keep `factor` -- and apply the random to all channels (a different random shift value for each channel).

Reply:
Done.